### PR TITLE
[CDAP-18674] Use serverside pagination in DeployedPipelineView

### PR DESCRIPTION
**Pull request by lahwran**
# Use serverside pagination in DeployedPipelineView

## Description
Remove local pagination and sorting code, switch to using serverside pagination and sorting code.

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-18674](https://cdap.atlassian.net/browse/CDAP-18674)

### CODE REVIEW

File: app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/index.tsx [hunk 1]
```diff
@@ -28,27 +27,16 @@ import './PipelineTable.scss';
...
 const PREFIX = 'features.PipelineList';
 
-const PipelineTableView: React.SFC<IProps> = ({
-  pipelines,
-  filteredPipelines: filteredList,
-  search,
-  onClear,
-  refetch,
-}) => {
+const PipelineTableView: React.SFC<IProps> = ({ pipelines, search, onClear, refetch }) => {
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/index.tsx:37]: i know you just removed a prop but in the future can you remove the : React.SFC<> in favor of `({}: IProps) =>`

File: app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/index.tsx [hunk 2]
```diff
@@ -77,7 +65,7 @@ const PipelineTableView: React.SFC<IProps> = ({
   return (
     <div className="grid-wrapper pipeline-list-table">
       <div className="grid grid-container">
-        <If condition={pipelines && filteredList && filteredList.length > 0}>
+        <If condition={pipelines && pipelines.length > 0}>
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/PipelineTable/index.tsx:68]: If you alter an `If` component in the future please remove it.

File: app/cdap/components/PipelineList/DeployedPipelineView/SearchBox/index.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ interface ISearchBoxProps {
 }
 const PREFIX = 'features.PipelineList';
 
-const SearchBoxView: React.SFC<ISearchBoxProps> = ({ value, onChange }) => {
+const SearchBoxView: React.FC<ISearchBoxProps> = ({ value, onChange }) => {
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/SearchBox/index.tsx:29]: let typescript infer type here

File: app/cdap/components/PipelineList/DeployedPipelineView/index.tsx [hunk 4]
```diff
@@ -15,66 +15,63 @@
...
 import { useQuery } from '@apollo/react-hooks';
+import { useSelector } from 'react-redux';
 import If from 'components/shared/If';
 import { categorizeGraphQlErrors } from 'services/helpers';
 import ErrorBanner from 'components/shared/ErrorBanner';
 import T from 'i18n-react';
+import { useDebounce } from 'services/react/customHooks/useDebounce';
 
 import './DeployedPipelineView.scss';
 const I18N_PREFIX = 'features.PipelineList.DeployedPipelineView';
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/index.tsx:42]: all your imports should go before a variable dec

File: app/cdap/components/PipelineList/DeployedPipelineView/index.tsx [hunk 5]
```diff
@@ -94,15 +91,91 @@ const DeployedPipeline: React.FC = () => {
...
+        return errs ? errs[0][0] : 'Unknown error';
       }
     }
   }
+  // no error, no message
+  return '';
+};
 
-  setFilteredPipelines(data.pipelines);
+const DeployedPipeline: React.FC = () => {
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/index.tsx:103]: typescript should infer this type

File: app/cdap/components/PipelineList/DeployedPipelineView/index.tsx [hunk 6]
```diff
@@ -94,15 +91,91 @@ const DeployedPipeline: React.FC = () => {
...
+            time
+          }
+        }
+        nextPageToken
+      }
+    }
+  `;
+
+  // on unmount
+  React.useEffect(() => {
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DeployedPipelineView/index.tsx:140]: You have {useEffect} already imported

File: app/cdap/components/PipelineList/DraftPipelineView/Pagination/index.tsx [hunk 7]
```diff
@@ -14,9 +14,67 @@
...
+  numPipelines: number;
+  pageLimit: number;
+  shouldDisplay: boolean;
+}
+
+const PaginationContainer = styled.div`
+  margin-right: 50px;
+`;
+
+const PaginationView: React.SFC<IPaginationProps> = ({
```
Comment (seanfeld) [inline app/cdap/components/PipelineList/DraftPipelineView/Pagination/index.tsx:35]: typescript should infer this type

File: graphql/Query/pipelinesResolver.js [hunk 8]
```diff
@@ -31,15 +31,29 @@ export async function queryTypePipelinesResolver(parent, args, context) {
...
   context.namespace = namespace;
 
   const errorModifiersFn = (error, statusCode) => {
     return new ApolloError(error, statusCode, { errorOrigin: 'pipelines' });
-  }
+  };
 
   const apps = await requestPromiseWrapper(options, context, null, errorModifiersFn);
-  return orderBy(apps, [(app) => app.name], ['asc']);
+  console.log("response: ", JSON.stringify(apps, null, 2), "for request", JSON.stringify(params), "to url", path);
```
Comment (seanfeld) [inline graphql/Query/pipelinesResolver.js]: do we need this console log?

**Review by seanfeld — APPROVED** [review]
I only have nits - if this is crazy time sensitive you can ignore them except is the console log needed for the finished code?

File: graphql/Query/pipelinesResolver.js (hunk 8 above)
Comment (lahwran) [reply graphql/Query/pipelinesResolver.js]: Fixed this one. I'll PR the others to develop in a few minutes.

**Pull request merged**